gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -307,105 +329,137 @@ protected IcebergTable validateGetFilePathsGivenDestState(
   }
 
   private CopyConfiguration createEmptyCopyConfiguration(FileSystem fs) {
-    return CopyConfiguration.builder(fs, copyConfigProperties)
-        .copyContext(new CopyContext())
-        .build();
+    return CopyConfiguration.builder(fs, copyConfigProperties).copyContext(new CopyContext()).build();
   }
 
   private static void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
-    List<String> actual = new ArrayList<>();
+    Set<String> actual = new HashSet<>();
     for (CopyEntity copyEntity : copyEntities) {
       String json = copyEntity.toString();
-      String filepath = new Gson().fromJson(json, JsonObject.class)
-          .getAsJsonObject("object-data").getAsJsonObject("origin")
-          .getAsJsonObject("object-data").getAsJsonObject("path")
-          .getAsJsonObject("object-data").getAsJsonObject("uri")
-          .getAsJsonPrimitive("object-data").getAsString();
+      String filepath = CopyEntityDeserializer.getFilePathAsStringFromJson(json);
       actual.add(filepath);
     }
-    Assert.assertEquals(actual.size(), expected.size(),
-        "Set" + actual.toString() + " vs Set" + expected.toString());
+    Assert.assertEquals(actual.size(), expected.size(), "Set" + actual.toString() + " vs Set" + expected.toString());
     Assert.assertEqualsNoOrder(actual.toArray(), expected.toArray());
   }
 
+  private static void verifyFsOwnershipAndPermissionPreservation(Collection<CopyEntity> copyEntities,
+      Map<Path, FileStatus> expectedPathsAndFileStatuses) {
+
+    for (CopyEntity copyEntity : copyEntities) {
+      String copyEntityJson = copyEntity.toString();
+
+      JsonArray ancestorOwnerAndPermissions = CopyEntityDeserializer.getAncestorOwnerAndPermissions(copyEntityJson);
+      String filePath = CopyEntityDeserializer.getFilePathAsStringFromJson(copyEntityJson);
+      CopyEntityDeserializer.DestinationOwnerAndPermissions destinationOwnerAndPermissions = CopyEntityDeserializer.getDestinationOwnerAndPermissions(copyEntityJson);
+
+      Assert.assertEquals(ancestorOwnerAndPermissions.size(), new Path(filePath).getParent().depth() - 1);
+      Assert.assertEquals(expectedPathsAndFileStatuses.get(new Path(filePath)).getOwner(),
+          destinationOwnerAndPermissions.owner);
+      Assert.assertEquals(expectedPathsAndFileStatuses.get(new Path(filePath)).getGroup(),
+          destinationOwnerAndPermissions.group);
+      Assert.assertEquals(expectedPathsAndFileStatuses.get(new Path(filePath)).getPermission().getUserAction().toString(),
+          destinationOwnerAndPermissions.userActionPermission);
+      Assert.assertEquals(expectedPathsAndFileStatuses.get(new Path(filePath)).getPermission().getGroupAction().toString(),
+          destinationOwnerAndPermissions.groupActionPermission);
+      Assert.assertEquals(expectedPathsAndFileStatuses.get(new Path(filePath)).getPermission().getOtherAction().toString(),
+          destinationOwnerAndPermissions.otherActionPermission);
```

Comment:
at the least, let's tidy up w/ an intermediate:
```
FileStatus expectedFileStatus = expectedPathsAndFileStatuses.get(new Path(filePath));
```
but even better might be to abstract:
```
void verifyDestFileStatus(CopyEntityDeserializer.DestinationOwnerAndPermissions observed, FileStatus expected) ...
```

Reply:
updated my PR with incorporating abstraction as suggested above